src/Microsoft.AspNetCore.Mvc.ApiExplorer/DefaultApiDescriptionProvider.cs

```diff
@@ -271,8 +284,46 @@ private IList<ApiParameterDescription> GetParameters(ApiParameterContext context
                     Source = BindingSource.Path,
                 });
             }
+        }
 
-            return context.Results;
+        internal static void ProcessIsRequired(ApiParameterContext context)
+        {
+            foreach (var parameter in context.Results)
+            {
+                if (parameter.Source == BindingSource.Body)
+                {
+                    parameter.IsRequired = true;
+                }
+
+                if (parameter.ModelMetadata != null && parameter.ModelMetadata.IsBindingRequired)
```

Comment:
When would `ModelMetadata` property be `null` e.g. is this only for route parameters that don't match an actual parameter (but are added to `Results`)?

Reply:
Unit tests